app/src/main/java/org/oppia/app/recyclerview/GridAutoFitLayoutManager.java

```diff
@@ -0,0 +1,57 @@
+package org.oppia.app.recyclerview;
+
+import android.content.Context;
+import android.util.TypedValue;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+/**
+ * This is used to compute the number of columns based on the predicted size of the recycler view (which should be the
+ * width of the parent given the match_parents) divided by the expected column size.
+ * for reference https://github.com/pushpalroy/talkie/blob/master/app/src/main/java/com/pushpal/talkie/model/util/GridAutoFitLayoutManager.java.
+ */
+public class GridAutoFitLayoutManager extends GridLayoutManager {
+  private int mColumnWidth;
+  private boolean mColumnWidthChanged = true;
+
+  public GridAutoFitLayoutManager(Context context, int columnWidth) {
+    /* Initially set spanCount to 1, will be changed automatically later. */
+    super(context, 1);
+    setColumnWidth(checkedColumnWidth(context, columnWidth));
+  }
+
+  public GridAutoFitLayoutManager(Context context, int columnWidth, int orientation, boolean reverseLayout) { /* Initially set spanCount to 1, will be changed automatically later. */
```

Comment:
I think this should be removed as we are not using it, also, if we need this in future then  actually your reference link can easily help to write this back.

Reply:
Removed.